std-bits/base/src/main/java/org/enso/base/Random_Utils.java

```diff
@@ -0,0 +1,38 @@
+package org.enso.base;
+
+import java.util.Arrays;
+import java.util.Random;
+
+public class Random_Utils {
+  /** Samples k random values from the input. */
+  public static Object[] sample(Object[] array, int k, Random rng) {
+    k = Math.min(k, array.length);
+    var copy = Arrays.copyOf(array, array.length);
+    shuffleFirstInPlace(copy, k, rng);
+    return Arrays.copyOf(copy, k);
+  }
+
+  public static Long[] random_indices(int n, int k, Random rng) {
+    /*
+     * TODO while acceptable for `k` close to `n`, for `k <<< n`, this algorithm is not efficient, a
+     * better one should be implemented, see: https://www.pivotaltracker.com/story/show/182853142
+     */
+    Long[] indices = new Long[n];
+    for (int i = 0; i < n; ++i) {
+      indices[i] = (long) i;
+    }
+    k = Math.min(k, n);
+    shuffleFirstInPlace(indices, k, rng);
+    return Arrays.copyOf(indices, k);
+  }
+
+  private static <T> void shuffleFirstInPlace(T[] array, int k, Random rng) {
```

Comment:
why `first`?

Reply:
Shuffles the first `k` elements of the array.

Reply:
Hm, given your question maybe the name I picked isn't best indeed.

What the function does is it reorders the element of the array in such a way that the first `k` elements are randomly chosen from the whole array - this allows to sample `k` random elements from the array.

@hubertp do you have a suggestion of what other name we may use? Maybe something like `sampleInPlace` will be better?